pkg/target/http.go

```diff
@@ -212,9 +213,77 @@ func AdaptHTTPTargetFunc(f func(c *HTTPTargetConfig) (*HTTPTarget, error)) HTTPT
 	}
 }
 
-func (ht *HTTPTarget) Write(messages []*models.Message) (*models.TargetWriteResult, error) {
+// When we have dynamic headers, batching by header must necessarily run first. This is a http specific function,
+// so defining it here and fixing it into the Write function avoids complexity in configuration
+func (ht *HTTPTarget) groupByDynamicHeaders(batches []*models.MessageBatch) ([]*models.MessageBatch, []*models.Message, []*models.Message) {
```

Comment:
I think i have an idea, but not sure exactly about what you mean by configuration complexity. Is it about the feature switch?  So far i'd still prefer batch transformations decoupled, so commenting to understand whether this could be a counter argument.

Reply:
We must group the data by dynamic headers before we create a request template. If we create the template before that then we cannot batch only events with the same headers together afterwards.

If we decouple this from the target, then the user has to understand this nuance of how the code works in order to configure the app correctly. Or to put it another way - they can have a valid configuration that fits the API that we have provided, which breaks a feature.

Does that make sense?

Reply:
Thanks for explaining @colmsnowplow ! It does make more sense. 
To confirm, the order before HTTP target writes is:
Group-by > Chunks > Templater
Is this correct? 
A side question: For the chuck size limits we won't take into account the template size. Is this ok?

Reply:
I don't see why we wouldn't take the template into account?